src/device-api/v2.ts

```diff
@@ -59,66 +59,64 @@ export function createV2Api(router: Router) {
 			return;
 		}
 
-		return applicationManager.lockingIfNecessary(appId, { force }, () => {
-			return Promise.all([applicationManager.getCurrentApps(), getApp(appId)])
-				.then(([apps, targetApp]) => {
-					const app = apps[appId];
-
-					if (app == null) {
-						res.status(404).send(appNotFoundMessage);
-						return;
-					}
+		return Promise.all([applicationManager.getCurrentApps(), getApp(appId)])
+			.then(([apps, targetApp]) => {
+				const app = apps[appId];
 
-					// Work if we have a service name or an image id
-					if (imageId == null) {
-						if (serviceName == null) {
-							throw new Error(v2ServiceEndpointInputErrorMessage);
-						}
-					}
+				if (app == null) {
+					res.status(404).send(appNotFoundMessage);
+					return;
+				}
 
-					let service: Service | undefined;
-					let targetService: Service | undefined;
-					if (imageId != null) {
-						service = _.find(app.services, (svc) => svc.imageId === imageId);
-						targetService = _.find(
-							targetApp.services,
-							(svc) => svc.imageId === imageId,
-						);
-					} else {
-						service = _.find(
-							app.services,
-							(svc) => svc.serviceName === serviceName,
-						);
-						targetService = _.find(
-							targetApp.services,
-							(svc) => svc.serviceName === serviceName,
-						);
-					}
-					if (service == null) {
-						res.status(404).send(serviceNotFoundMessage);
-						return;
+				// Work if we have a service name or an image id
+				if (imageId == null) {
+					if (serviceName == null) {
+						throw new Error(v2ServiceEndpointInputErrorMessage);
 					}
+				}
+
+				let service: Service | undefined;
+				let targetService: Service | undefined;
+				if (imageId != null) {
+					service = _.find(app.services, (svc) => svc.imageId === imageId);
+					targetService = _.find(
+						targetApp.services,
+						(svc) => svc.imageId === imageId,
+					);
+				} else {
+					service = _.find(
+						app.services,
+						(svc) => svc.serviceName === serviceName,
+					);
+					targetService = _.find(
+						targetApp.services,
+						(svc) => svc.serviceName === serviceName,
+					);
+				}
+				if (service == null) {
+					res.status(404).send(serviceNotFoundMessage);
+					return;
+				}
 
-					applicationManager.setTargetVolatileForService(service.imageId!, {
-						running: action !== 'stop',
+				applicationManager.setTargetVolatileForService(service.imageId!, {
+					running: action !== 'stop',
+				});
+				return applicationManager
+					.executeStep(
+						generateStep(action, {
+							current: service,
+							target: targetService,
+							wait: true,
+						}),
+						{
+							force,
+						},
+					)
+					.then(() => {
+						res.status(200).send('OK');
 					});
-					return applicationManager
-						.executeStep(
-							generateStep(action, {
-								current: service,
-								target: targetService,
-								wait: true,
-							}),
-							{
-								skipLock: true,
```

Comment:
What's the effective difference between using force now compared to skiplock here?

Reply:
`skipLock` ignores the lock and calls the funcion directly, `force` checks for a lock,  removes it if exists, and then sets its own lock to perorm the function call. In this case, since the `lockingIfNecessary` call above had the lock, it was ok to skip the lock for the `executeStep` function